onnx/defs/math/defs.cc

```diff
@@ -821,14 +816,120 @@ ONNX_OPERATOR_SET_SCHEMA(
 ONNX_OPERATOR_SET_SCHEMA(
     Softmax,
     13,
-    OpSchema().FillUsing(
-        SoftmaxFamilyDocGenerator("softmax", "normalized exponential")));
+    OpSchema()
+        .FillUsing(
+            SoftmaxFamilyDocGenerator("softmax", "normalized exponential"))
+        .SetContextDependentFunctionBodyBuilder(
+            [](const FunctionBodyBuildContext& ctx,
```

Comment:
did you verify the correctness of the subgraph?  (the "expanded" and non "expanded" model generated in this PR, feeding them same inputs will get same outputs)

Reply:
Shouldn't this functional decomposition include a Flatten call? (the part that coerces the shape to 2D)

Here's the decomposition I wrote down before:
https://fdwr.github.io/LostOnnxDocs/OperatorFormulas.html

```
function SoftMax2D(Input; Output; axis):
  MaxInput = ReduceMax(Input, axes=[1], keepdims=1)
  ExpInput = Exp(Input - MaxInput)
  ReducedExpInput = ReduceSum(ExpInput, axis=[1], keepdims=1)
  Output = ExpInput / ReducedExpInput
endfunction

function F(Input; Output; axis):
  FlattenedInput = Flatten(Input, axis) // Flatten to 2D
  NormalizedInput = SoftMax2D(FlattenedInput)
  Output = Reshape(NormalizedInput, Shape(X))
endfunction
```

Reply:
@linkerzhang @fdwr I have verified the correctness of the subgraph by inferencing the subgraph in onnxruntime and comparing the result with the reference numpy implementation